agent/engine/task_manager.go

```diff
@@ -161,7 +181,7 @@ func (mtask *managedTask) overseeTask() {
 		// Conversely, for it to spin in steady state it will have to have been
 		// loaded in steady state or progressed through here, so saving here should
 		// be sufficient to capture state changes.
-		err := mtask.engine.saver.Save()
+		err := mtask.saver.Save()
```

Comment:
It looks like the error returned by the `Save()` isn't critical, do you think it makes sense to move the error check into the `Save()` itself or we don't even care this error like other places?

Reply:
The interface itself specifies it as a method that returns an error. That can probably remain the way it is. I have however modified it so that the task arn gets printed here as well and we have more context around this error